site/docs/hosting/cloudflare-workers.md

```diff
@@ -0,0 +1,120 @@
+# Hosting: Cloudflare Workers
+
+[Cloudflare Workers](https://workers.cloudflare.com/) is a public serverless computing service developed by Cloudflare, which allows you to run JavaScript in the edge.
+Specifically, it is a worker which handles HTTP traffic and is written against the [Service Worker API](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API).
+Together, with other services that are also provided by Cloudflare, namely:
+
+1. [Workers KV](https://developers.cloudflare.com/workers/runtime-apis/kv/), a global, low-latency, key-value data store,
+2. [Durable Objects](https://developers.cloudflare.com/workers/runtime-apis/durable-objects/), a low-latency coordination and consistent storage for the Workers,
+3. [R2](https://developers.cloudflare.com/r2/), an S3-compatible, zero egress-fee, globally distributed object storage,
+4. [D1](https://developers.cloudflare.com/d1/), a queryable relational database at the edge,
+5. [Queues](https://developers.cloudflare.com/queues/), a message queuing service that allows applications to reliably send and receive messages using Cloudflare Workers.
+
+we can easily create our own Telegram bots, as well as [Web Apps](https://core.telegram.org/bots/webapps), all on the edge, and even for free (with some quotas)!
+
+This guide shows you how you can host your grammY bots on Cloudflare Workers.
+
+## Prerequisites
+
+To follow along, please make sure that you have a Cloudflare account with your workers subdomain configured.
+
+## Setting Things Up
+
+First, create a new project:
+
+```sh
+npx wrangler generate my-bot
+```
+
+You can change `my-bot` to whatever you want.
+This will be the name of your bot and the project directory.
+
+After running the above command, follow the instructions you see to initialize the project.
+There, you can choose between JavaScript or TypeScript.
+
+When the project is initialized, cd into `my-bot` or whatever directory you initialized your project in.
+Depending on how you initialized the project, you should see a file structure similar to the following:
+
+```sh
+.
+|-- node_modules
+|-- package-lock.json
+|-- package.json
+|-- src
+│   └── index.test.js
+│   └── index.js
+`-- wrangler.toml
+```
```

Comment:
```suggestion
```asciiart:no-line-numbers
.
├── node_modules
├── package.json
├── package-lock.json
├── src
│   ├── index.js
│   └── index.test.js
└── wrangler.toml
```
```

Reply:
Should we maybe encourage people to use TypeScript, and use the corresponding example structure?